src/main/java/com/mraof/minestuck/block/redstone/RetractableSpikesBlock.java

```diff
@@ -0,0 +1,183 @@
+package com.mraof.minestuck.block.redstone;
+
+import com.mraof.minestuck.effects.CreativeShockEffect;
+import com.mraof.minestuck.util.MSDamageSources;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.MobEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.pathfinding.PathNodeType;
+import net.minecraft.state.BooleanProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.util.*;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraftforge.common.util.Constants;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Acts similar to spike blocks, except the damage is dependent on whether the spikes are extended or not. Spikes will extend if the block is powered or if the block is in Pressure Sensitive mode and a mob has stepped on it.
+ */
+public class RetractableSpikesBlock extends Block
+{
+	public static final BooleanProperty POWERED = BlockStateProperties.POWERED;
+	public static final BooleanProperty PRESSURE_SENSITIVE = BlockStateProperties.ENABLED;
+	
+	public RetractableSpikesBlock(Properties properties)
+	{
+		super(properties);
+		this.registerDefaultState(stateDefinition.any().setValue(POWERED, false).setValue(PRESSURE_SENSITIVE, false));
+	}
+	
+	@Override
+	public void fallOn(World worldIn, BlockPos pos, Entity entityIn, float fallDistance)
+	{
+		if(worldIn.getBlockState(pos).getValue(POWERED) && fallDistance > 3)
+		{
+			if(entityIn instanceof LivingEntity)
```

Comment:
For the instanceof check, I was worried about the deletion of item entities on falling on it. As for the super call, what is your suggested change?

Reply:
I was suggesting something along the line of `if (...) entityIn.causeFallDamage(...); else super.fallOn(...);` so that you call one or the other, and never both. Also, I don't think you need to worry about items when it comes to fall damage, but I can look into it some more tonight.